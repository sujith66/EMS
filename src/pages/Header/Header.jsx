import * as React from "react";
import AppBar from "@mui/material/AppBar";
import { useNavigate } from "react-router-dom";

export default function Header() {
  const navigate = useNavigate();

  return (
    <AppBar
      sx={{
        display: "flex",
        justifyContent: "center",
        height: "10%",
        paddingLeft: "2rem",
      }}
      onClick={() => navigate("/")}
    >
      Endpoints Manager
    </AppBar>
  );
}
