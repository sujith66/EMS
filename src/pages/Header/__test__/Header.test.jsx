import React from "react";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import Header from "../Header";

const mockHeader = ()=>(    <Header />);
//mocking useNavigate
const mockedUsedNavigate = jest.fn();

jest.mock("react-router-dom", () => ({
  ...jest.requireActual("react-router-dom"),
  useNavigate: () => mockedUsedNavigate,
}));

//Test suite which will contain all the tests for Header
describe("Testing Header", () => {
  test("check if the  Header component is rendered successfully", () => {
    render(mockHeader());
    expect(screen.getByText("Endpoints Manager")).toBeInTheDocument();
  });

  test("check if Header component has failed to render", () => {
    render(mockHeader());
    expect(screen.queryByText("Endpoints Manager1")).not.toBeInTheDocument();
  });

  test("matches the snapshot to check if the component renders correctly", () => {
    const tree = renderer
      .create((mockHeader()))
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
