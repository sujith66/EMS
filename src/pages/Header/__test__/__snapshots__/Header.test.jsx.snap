// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Header matches the snapshot to check if the component renders correctly 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionFixed mui-fixed css-1lir7g8-MuiPaper-root-MuiAppBar-root"
  onClick={[Function]}
>
  Endpoints Manager
</header>
`;
